src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -92,13 +94,16 @@
       File="$(AssemblyInfoFile)"
       Lines="@(AssemblyInfoUsings);@(AssemblyInfoLines)"
       Overwrite="true" />
-
-    <ItemGroup>
-      <Compile Include="$(AssemblyInfoFile)" />
-      <FileWrites Include="$(AssemblyInfoFile)" />
-    </ItemGroup>
   </Target>
 
+  <ItemGroup Condition="'$(GenerateAssemblyInfo)'=='true'">
+    <Compile Include="$(AssemblyInfoFile)">
```

Comment:
After further testing, it looks like if the "bin" directory is empty, _AssemblyInfo.cs is not generated for any of the projects until they are built.  When I was testing it before, I was just deleting the project's directory in "bin\obj", which is where _AssemblyInfo.cs was used, but perhaps the system was getting the file from somewhere else in some cases.